### PR TITLE
Add negative test cases

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
$Subject

Fixes: https://github.com/ballerina-platform/ballerina-standard-library/issues/2300
## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: ballerina/tests/error.bal [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+@test:Config {
+    groups: ["error"]
+}
+function TestConversionError() returns error? {
+    sql:DateValue value = new("hi");
+    sql:ParameterizedQuery sqlQuery = `SELECT string_type from DataTable WHERE row_id = ${value}`;
+    Client dbClient = check new (host, user, password, errorDB, port);
+    string|error stringVal = dbClient->queryRow(sqlQuery);
+    check dbClient.close();
+    //test:assertTrue(stringVal is sql:ConversionError);
```
Comment (niveathika) [inline ballerina/tests/error.bal:183]: Can this be removed?

File: ballerina/tests/error.bal [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+        DROP TABLE IF EXISTS DataTable;
+
+        CREATE TABLE DataTable(
+            row_id       INT,
+            string_type  VARCHAR(50),
+            PRIMARY KEY (row_id)
+        );
+
+        INSERT INTO DataTable (row_id, string_type) VALUES(1, '{""q}');
+        `;
```
Comment (kaneeldias) [inline ballerina/tests/error.bal:40]:
```suggestion
    `;
```

File: ballerina/tests/error.bal [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+}
+function TestConnectionClose() returns error? {
+    sql:ParameterizedQuery sqlQuery = `SELECT string_type from DataTable WHERE row_id = 1`;
+    Client dbClient = check new (host, user, password, errorDB, port);
+    check dbClient.close();
+    string|error stringVal = dbClient->queryRow(sqlQuery);
+    test:assertTrue(stringVal is sql:ApplicationError);
+    sql:ApplicationError sqlerror = <sql:ApplicationError> stringVal;
+    test:assertEquals(sqlerror.message(), "SQL Client is already closed, hence further operations are not allowed",
+                sqlerror.message());
```
Comment (kaneeldias) [inline ballerina/tests/error.bal:88]: Do we need this?

File: ballerina/tests/error.bal [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+    groups: ["error"]
+}
+function TestInvalidTableName() returns error? {
+    sql:ParameterizedQuery sqlQuery = `SELECT string_type from Data WHERE row_id = 1`;
+    Client dbClient = check new (host, user, password, errorDB, port);
+    string|error stringVal = dbClient->queryRow(sqlQuery);
+    check dbClient.close();
+    test:assertTrue(stringVal is sql:DatabaseError);
+    error sqlerror = <error> stringVal;
+     test:assertEquals(sqlerror.message(), "Error while executing SQL query: SELECT string_type from Data " +
```
Comment (kaneeldias) [inline ballerina/tests/error.bal:101]:
```suggestion
    test:assertEquals(sqlerror.message(), "Error while executing SQL query: SELECT string_type from Data " +
```

File: ballerina/tests/error.bal [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+                "WHERE row_id = 1. Invalid object name 'Data'..", sqlerror.message());
```
Comment (kaneeldias) [inline ballerina/tests/error.bal:102]:
```suggestion
                "WHERE row_id = 1. Invalid object name 'Data'..");
```

File: ballerina/tests/error.bal [hunk 6]
```diff
@@ -0,0 +1,308 @@
...
+    test:assertTrue(strings:includes(sqlError.message(), "No mapping field found for SQL table column 'string_type' " +
+          "in the record type 'stringValue'"), sqlError.message());
+}
+
+@test:Config {
+    groups: ["error"]
+}
+function TestIntegrityConstraintViolation() returns error? {
+    Client dbClient = check new (host, user, password, errorDB, port);
+    _ = check  dbClient->execute(`CREATE TABLE employees( employee_id int not null,
```
Comment (kaneeldias) [inline ballerina/tests/error.bal:242]:
```suggestion
    _ = check dbClient->execute(`CREATE TABLE employees( employee_id int not null,
```

File: ballerina/tests/error.bal [hunk 7]
```diff
@@ -0,0 +1,308 @@
...
+                                                        employee_name varchar (75) not null,supervisor_name varchar(75),
+                                                        CONSTRAINT employee_pk PRIMARY KEY (employee_id))`);
+    _ = check  dbClient->execute(`CREATE TABLE departments( department_id int not null,employee_id int not
```
Comment (kaneeldias) [inline ballerina/tests/error.bal:245]:
```suggestion
    _ = check dbClient->execute(`CREATE TABLE departments( department_id int not null,employee_id int not
```

**Review by niveathika — APPROVED** [review]

**Pull request merged**